components/compositing/compositor.rs

```diff
@@ -1143,11 +1296,35 @@ impl<Window: WindowMethods + ?Sized> IOCompositor<Window> {
             })
     }
 
-    fn create_pipeline_details_for_frame_tree(&mut self, frame_tree: &SendableFrameTree) {
-        self.pipeline_details(frame_tree.pipeline.id).pipeline = Some(frame_tree.pipeline.clone());
+    fn update_pipeline_details_tree(
```

Comment:
This has the sneaky side-effect of creating `PipelineDetails` if they don't exist, so I think the method name is really helpful to reveal this subtlety.

```suggestion
    fn create_or_update_pipeline_details_with_frame_tree(
```

Reply:
done in b2528dc1bbe43ea18822c51cc3c2d6c10222bc0e